### DISCUSSION TITLE
Recurring Tasks

### DISCUSSION THREAD

**Discussion by varlevi** [alainm23/planify#661] [Ideas]
Can we talk about recurring tasks? I know that there is some support when integrating with Todoist, but there appears to be little support when just running the application locally and there are also some problems with Todoist support. I think many users would find recurring tasks to be very helpful, including myself. What would it take to add fully-featured recurring tasks to Planner both when running locally and when running and integrating with Todoist?

**Comment by alainm23** [comment]
Thank you for your comment. Planner's recurring tasks are based on Todoist and using natural language to make it work. 
e.g.
- every day
- every week

At the same time this varies depending on the language used in Todoist (Spanish):
- cada dia
- cada semana

There're even complicated variations:
- every 27th (cada 27)
- every 3rd friday (cada tercer viernes)
- every last day (cada fin de mes)
- every day starting aug 3 (cada día empezando el 3 ago)

I still have no idea how to process natural language, probably the solution is to use 'Regular expressions'. I hope to receive help or a solution for this problem, in the meantime I'll continue working and investigating to solve this problem.

**Comment by alainm23** [comment]
I just found a [library](https://github.com/wanasit/chrono/) that could help us with this problem.


https://user-images.githubusercontent.com/33765137/103604184-5888d200-4ede-11eb-942e-03e642e5acef.mp4

**Comment by alainm23** [comment]
This feature is available in Planify 4.1. However, if you use Todoist, bidirectional sync is not yet available.
